sunpy/coordinates/spice.py

```diff
@@ -467,3 +467,72 @@ def get_fov(instrument, time, *, resolution=100):
+    """
+    Computes the rotation matrix that converts vectors from the ``source_frame`` to the ``target_frame``
+    at the specified time using the SPICE toolkit.
+
+    Parameters
+    ----------
+    source_frame : `str`, `int`, `~astropy.coordinates.BaseCoordinateFrame`
+        The frame of the input vector field, specified as a frame name, SPICE ID, or Astropy BaseCoordinateFrame.
+
+    target_frame : `str`, `int`, `~astropy.coordinates.BaseCoordinateFrame`
+        The frame to which the vector field will be transformed, specified in the same formats as ``source_frame``.
+
+    from_time : {parse_time_types}
+        The time at which the vector field is defined in the source frame, in any format accepted by `sunpy.time.parse_time`.
+
+    to_time : {parse_time_types}
+        The time at which the vector field should be defined in the target frame. Defaults to ``from_time``, resulting in a spatial-only transformation.
+
+    Returns
+    -------
+    `~numpy.ndarray`
+        A 3x3 rotation matrix that represents the rotation between the two frames.
+
+    Examples
+    --------
+    >>> from sunpy.coordinates.spice import get_rotation_matrix
+    >>> source_frame = "J2000"
+    >>> target_frame = "Galactic"
+    >>> from_time = '2001-01-01T00:00:00'
+
+    >>> rotation_matrix = get_rotation_matrix(source_frame, target_frame, from_time)
+
+    * The `get_rotation_matrix` function will first determine the rotation matrix that converts vectors from the source frame to the target frame at the specified time.
+    * This rotation matrix is a 3x3 matrix that represents the rotation between the two frames.
+
+    >>> rotation_matrix
+    array([[-0.05487554, -0.8734371 , -0.48383499],
+           [ 0.49410945, -0.44482959,  0.74698225],
+           [-0.86766614, -0.19807639,  0.45598379]])
+
+    * ``rotation_matrix`` can now be used with the vector components to transform the field. An example way is shown below.
+
+    >>> vec_components = [1, 0, 0] * u.T
+    >>> transformed_matrix = rotation_matrix @ vec_components
+    >>> transformed_matrix
+    <Quantity [-0.05487554, 0.49410945, -0.86766614] T>
+
+    """
+    # Convert sunpy frames to SPICE strings
+    source_frame_spice = source_frame.name.upper() if hasattr(source_frame, 'name') else str(source_frame).upper()
+    target_frame_spice = target_frame.name.upper() if hasattr(target_frame, 'name') else str(target_frame).upper()
+
+    from_time = parse_time(from_time)
+    to_time = parse_time(to_time) if to_time else from_time
+
+    from_time_et = _convert_to_et(from_time)
+    to_time_et = _convert_to_et(to_time)
+
+    # Compute state transformation matrix using spiceypy
+    state_matrix = spiceypy.sxform(source_frame_spice, target_frame_spice, to_time_et or from_time_et)
+
+    # The state transformation matrix is a 6x6 matrix. The upper left 3x3 block is the rotation matrix.
+    rotation_matrix = state_matrix[:3, :3]
```

Comment:
This is incorrect because you are not transforming between the two times, but rather always just using `to_time`.  The CSPICE/SpiceyPy API is limiting here, because only `pxfrm2` accepts two different times while only `sxform` accepts a time array.  So, what you'll need to do here is to make *two* sxform calls, transforming from `source_frame_spice` to SPICE's "J2000" frame and from "J2000" to `target_frame_spice`, and then multiply those two matrices.

Reply:
Thanks for the correction! I have incorporated the suggested changes.